### ISSUE TITLE
Isolate Memory Limit Segmentation Fault

### ISSUE THREAD

**Issue by josalhor** [cms-dev/cms#1123]
**Description:** I've been trying to create a problem and submit and answer to test the system. To my surprise the compilation never ended. After checking the logs I discovered the job never fully compiles (even though it is a python solution but I understood what it meant). the files in the `/tmp/`folder have the following content:

commands.log:
`isolate --box-id=90 --cg --cg-timing --chdir=/tmp --dir=/tmp=/tmp/cms-compile-1rdpgt0o/home:rw --dir=/etc/alternatives=/etc/alternatives --dir=/etc=/etc --dir=/var/lib/ghc=/var/lib/ghc --full-env --env=HOME=/tmp --cg-mem=524288 --stdout=/tmp/compilation_stdout_0.txt --processes=1000 --stderr=/tmp/compilation_stderr_0.txt --time=10 --wall-time=21 --meta=/tmp/cms-compile-1rdpgt0o/run.log.0 --run -- /usr/bin/python3 -m compileall -b .`

run.log.0:
`time:0.000
time-wall:0.182
max-rss:716
csw-voluntary:7
csw-forced:0
exitsig:11
status:SG
message:Caught fatal signal 11`

This seems to be relate to #1074 and https://github.com/ioi/isolate/issues/64 

**Steps to reproduce:**
I've not done anything special with my setup. Just follow documentation installation procedure, setup contest and task, send solution, get the behavior above. I tried running the isolate command, as expected I got the Caught fatal signal 11. After checking issue 64 of isolate I tried running isolate without any memory limit and everything worked as expected.

In compilation.py I found `sandbox.address_space = config.compilation_sandbox_max_memory_kib`So I went ahead and changed `self.compilation_sandbox_max_memory_kib = None` in conf.py everything worked as expected, since `IsolateSandbox` in `build_box_options` seems to take into account that case.

### Logs

Log Error: https://gist.github.com/josalhor/c66a68b4d9351ffb7aa217a4e29c0754

### System Information

**CMS version:** 1.4rc1
**Was CMS installed:** executed with absolute path
**Using a virtualenv:** yes
**Distribution and version:** Ubuntu 18.10 (although running KDE desktop, but I don't think that is relevant in this case).

**Comment by stefano-maggiolo** [comment]
So I guess it would be a matter of using isolate 1.8?

We are still a couple of versions behind because IIRC 1.7 required us to do a few small changes. I'll take a look if I can upgrade.

**Comment by josalhor** [comment]
My bad, I forgot to mention one other thing in my report:

I tried to update isolate with the latest version on master and it still wasn't working as expected. However I'm currently unsure if it is picking up my latest compiled version that I setup manually or if I should've recompiled using your script, so it is possible it is still using the version from the previous compilation somehow.

**Comment by wil93** [comment]
You can try running:

```
./prerequisites.py build_isolate
sudo ./prerequisites.py install_isolate
```

to make sure

**Comment by josalhor** [comment]
Yup, you're right, version 1.8 fixes it, rebuilding with wil93 commands fixed the issue.

**Comment by josalhor** [comment]
@stefano-maggiolo Did you have time to look at the necessary changes? Is there any security concern in running isolate 1.6?

I guess the changes have something to do with this commit? https://github.com/ioi/isolate/commit/715add21d25e066df6420d80917b954432f74d6b

I'm deploying CMS on a cluster, so I need maximum security.

**Comment by wil93** [comment]
News?

**Comment by josalhor** [comment]
Our project has been put on hold for various reasons.
Right now I cannot be of much help, I think this issue should be closed and another one about isolate 1.8 should be opened, as the cause of the issue at hand was found and attributed to isolate 1.6.
